pages/tokens.py

```diff
@@ -3,22 +3,25 @@
 
 import pandas as pd
 
+import dash
 import dash_bootstrap_components as dbc
 from dash import dcc, html, Input, Output, callback, State, register_page, dash_table
 
 from emission.storage.decorations.token_queries import insert_many_tokens
 import emission.core.get_database as edb
+import base64
+import io
 
 from utils.generate_qr_codes import saveAsQRCode
 from utils.generate_random_tokens import generateRandomTokensForProgram
 from utils.permissions import get_token_prefix, has_permission
 
+token_deletion_enabled = os.getenv('TOKEN_DELETION_ENABLED', 'False').lower() == 'true'
 
 if has_permission('token_generate'):
```

Comment:
we should also show the token page only when tokens are not autogenerated. If tokens are autogenerated, they don't need to be stored in the token_db and don't need to be added or deleted.